### PR TITLE
build: Skip custom vet check by default

### PR DESCRIPTION
**Pull request by jrajahalme**
Change the custom vet check flag (SKIP_CUSTOMVET_CHECK) default value from `false` to `true`. This allows Dev VM to start up. Without this change the Dev VM startup fails like this:

```
    runtime1: contrib/scripts/custom-vet-check.sh
    runtime1: main: ./... matched no packages
    runtime1: exit status 1
    runtime1: Makefile:511: recipe for target 'precheck' failed
    runtime1: make: *** [precheck] Error 1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
Custom vet check was already disabled for CI builds:
```
test/provision/compile.sh:    sudo -u vagrant -H -E make SKIP_CUSTOMVET_CHECK=true LOCKDEBUG=1 SKIP_K8S_CODE_GEN_CHECK=false SKIP_DOCS=true
```
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: Makefile [hunk 1]
```diff
@@ -47,7 +47,7 @@ LOGLEVEL ?= "error"
 SKIP_VET ?= "false"
 SKIP_KVSTORES ?= "false"
 SKIP_K8S_CODE_GEN_CHECK ?= "true"
-SKIP_CUSTOMVET_CHECK ?= "false"
+SKIP_CUSTOMVET_CHECK ?= "true"
```
Comment (joestringer) [inline Makefile:50]: This looks like it will disable the vet check for CI as well (`make precheck` in `.github/workflows/go-check.yaml`). If this is targeted to the dev VM, can we either fix the dev VM itself or change the dev VM startup scripting to address this?

**Review by joestringer — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (jrajahalme) [reply Makefile:50]: Good point, I have no idea what causes the failure, so I'll address this in the dev VM startup scripting instead!

**Comment by johngv2** [comment]
Hi guys, is this problem solved? I had the same problem and I would like to share my thoughts about.

**TL;DR:** probably cilium was not cloned inside $GOPATH.

**Why this error occurs?** 

After a long sequence of calls...
contrib/scripts/custom-vet-check.sh
tools/customvet
main() (vendor/github.com/cilium/customvet/main.go)
init()  (vendor/github.com/cilium/customvet/analysis/timeafter/time_after.go)
...
Run() (golang.org/x/tools/go/analysis/internal/checker/checker.go)
load() (golang.org/x/tools/go/analysis/internal/checker/checker.go)
           
the execution reaches Config.Load (golang.org/x/tools/go/analysis/internal/checker/checker.go) which returns an empty list of packages. Receiving this empty list, load() returns an error with the message "%s matched no packages".

Hope this helps someone ;)
